Assets/Scripts/States/ShootingState.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Tanks.Bullet;
+using UnityEngine;
+
+namespace Tanks.States
+{
+    public class ShootingState : TankState
+    {
+        bool Enterstate;
```

Comment:
why do you need this enter state?

Reply:
the component is disabled when you exit the state -  so update won't get called anyways!